packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -148,18 +163,28 @@ class AnalyzeCommand extends FlutterCommand {
 
     if (argResults['current-package']) {
       // ./lib/main.dart
-      String mainPath = 'lib/main.dart';
+      String mainPath = path.join('lib', 'main.dart');
       if (FileSystemEntity.isFileSync(mainPath)) {
         dartFiles.add(mainPath);
         pubSpecDirectories.add('.');
         foundAnyInCurrentDirectory = true;
       }
+
+      // Check for <package>/lib/<package>.dart.
+      String packageName = path.basename(Directory.current.path);
+      String mainFilePath = path.join('lib', '$packageName.dart');
```

Comment:
@Hixie, would you like me to remove this specific check (`<package>/lib/<package>.dart`), and keep the first one (`.../packages/*/lib/packageName.dart`)? Or do you not want to use a package name main lib convention at all? I can close this PR out if you have reservations -